sources/logdog/build.rs

```diff
@@ -1,11 +1,20 @@
-// Automatically generate README.md from rustdoc.
+// Automatically generate README.md from rustdoc and generate variant symlink
 
-use std::env;
 use std::fs::File;
 use std::io::Write;
-use std::path::PathBuf;
+use std::os::unix::fs::symlink;
+use std::path::{Path, PathBuf};
+use std::{env, fs, io, process};
+
+// The VARIANT variable is originally BUILDSYS_VARIANT, set in the top-level Makefile.toml,
+// and is passed through as VARIANT by the top-level Dockerfile.  It represents which OS
+// variant we're building, and therefore which API model to use.
+const ENV_VARIANT: &str = "VARIANT";
 
 fn main() {
+    println!("cargo:rerun-if-env-changed={}", ENV_VARIANT);
+    symlink_variant();
+
     // Check for environment variable "SKIP_README". If it is set,
```

Comment:
nit: can we do like `models` and put variant/readme code in separate functions, and just call those from main?

Reply:
[aa7d592](https://github.com/bottlerocket-os/bottlerocket/compare/60e5d3c8e65c58170e8f72d944a80be11ea098e0..aa7d59202dad5161a9dc8aff3995f1fbe0d4c706)